paddle/fluid/platform/nccl_helper.h

```diff
@@ -106,7 +106,7 @@ struct NCCLContextMap {
     }
     std::unique_ptr<ncclComm_t[]> comms(new ncclComm_t[order_.size()]);
     // if num_trainers == 1, should create a new nccl id for local comms.
-    if (num_trainers == 1) {
+    if (num_trainers == 1 && nccl_id == nullptr) {
```

Comment:
why change this? Is this a bug fix?

Reply:
Not a bug, ParallelGraph should initialize NCCl by ranks mode with the same `nccl_id`.